tests/test_project.py

```diff
@@ -655,7 +655,6 @@ def test_job_move(self):
         job = project_a.open_job(dict(a=0))
         job_b = project_b.open_job(dict(a=0))
         assert job != job_b
-        assert hash(job) != hash(job_b)
```

Comment:
So this test fails under the new changes since it only failed since the working directories for `job` and `job_b` were different

Reply:
Yes, this fails because of the relaxed definition of the `hash` function. `job == job_b` implies that `hash(job) == hash(job_b)`, but the logical inverse does not hold. In the Python data model, it is fine for objects to be non-equal and have the same hash.